kedro-org/kedro-viz
# Setup E2E Tests with Cypress

## Description

Resolves #1392 

## Development notes

- Added Cypress Configuration
- Created skeleton folder for cypress e2e tests
- Created init spec to discover various features offered by cypress
- Added test cases for global-toolbar as part of exploration
- Added data-cy attribute to React UI components - IconButton, Button to follow best practices recommendation from Cypress
- Removed few unused variables as part of lint check warnings

## QA notes

- Cypress tests added
- For headless mode and to run cypress tests locally execute command `npm run cy:test`
- To view cypress dashboard, run frontend and backend application individually using `npm start`, `make run` respectively and then execute the commands `npm run cy:dev`

NOTE: To execute cypress tests locally, you should have pre-installed all the Kedro-Viz dependencies mentioned in [contributing](/CONTRIBUTING.md)

## Checklist



<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1398"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
